src/libNetwork/P2PComm.cpp

```diff
@@ -148,6 +146,104 @@ void P2PComm::SendMessageCore(const Peer& peer,
     }
 }
 
+namespace
+{
+    bool readMsg(vector<unsigned char>* buf, int cli_sock, const Peer& from,
+                 const uint32_t message_length)
+    {
+        // Read the rest of the message
+        assert(buf);
+        buf->resize(message_length);
+        uint32_t read_length = 0;
+        uint32_t retryDuration = 1;
+
+        while (read_length != message_length)
```

Comment:
In the future I think maybe we should use a stronger check here (read_length < message_length)

Reply:
OK, I'm good to make modification in this PR